### PR TITLE
gasnet: Convert to AutotoolsPackage, update

### PR DESCRIPTION
**Pull request by eschnett**
- convert to AutotoolsPackage
- add new version 1.28.0
- create variants for IB and MPI support

### CODE REVIEW

File: var/spack/repos/builtin/packages/gasnet/package.py [hunk 1]
```diff
@@ -35,26 +35,30 @@ class Gasnet(Package):
...
     url      = "http://gasnet.lbl.gov/GASNet-1.24.0.tar.gz"
 
+    version('1.28.0', 'b44446d951d3d8954aa1570e3556ba61')
     version('1.24.0', 'c8afdf48381e8b5a7340bdb32ca0f41a')
 
-    def install(self, spec, prefix):
-        # TODO: don't use paths with @ in them.
-        change_sed_delimiter('@', ';', 'configure')
+    variant('ibv', default=False)
+    variant('mpi', default=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gasnet/package.py]: Add descriptions.

File: var/spack/repos/builtin/packages/gasnet/package.py [hunk 2]
```diff
@@ -35,26 +35,30 @@ class Gasnet(Package):
...
 
-        configure(
-            "--prefix=%s" % prefix,
+    depends_on('mpi', when='+mpi')
+
+    def configure_args(self):
+        args = [
+            "--prefix=%s" % self.spec.prefix,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/gasnet/package.py]: This is already added by `AutotoolsPackage`, no need to add it twice.

Comment (eschnett) [inline var/spack/repos/builtin/packages/gasnet/package.py]: Removed.

File: var/spack/repos/builtin/packages/gasnet/package.py (hunk 1 above)
Comment (eschnett) [inline var/spack/repos/builtin/packages/gasnet/package.py]: Added.

**Pull request merged**